test/root/requestHandlerTests.js

```diff
@@ -0,0 +1,394 @@
+/*!
+ * (c) Copyright 2020 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+/* eslint-env mocha */
+
+const assert = require('assert')
+const RequestHandler = require('../../lib/root/requestHandler')
+const nock = require('nock')
+const nockServerUrl = 'https://myNockServer.com'
+const uri = '/some/uri'
+const user = 'MyUser'
+const password = 'MyPassword'
+const UNAUTHORIZED_ERROR = {
+  name: 'StatusCodeError',
+  statusCode: 401
+}
+
+describe('requestHandler', function () {
+  this.slow(350)
+  let requestHandler
+  beforeEach(() => {
+    const requestHandlerInitialization = { user: user, password: password, server: nockServerUrl }
+    requestHandler = new RequestHandler(requestHandlerInitialization)
+  })
+
+  async function throwsUnauthorizedException (promise) {
+    await throwsException(promise, UNAUTHORIZED_ERROR)
+  }
+  async function throwsException (promise, exception) {
+    await assert.rejects(promise, exception)
+  }
+  function wrapInErrorMessage (message) {
+    return { message: 'Error: ' + message }
+  }
+
+  describe('#authenticate', () => {
+    it('makes a successful authentication request with the given username and password', async () => {
+      const scope = BuildServerResponses()
+        .withSuccessfulAuthentication(user, password)
+        .build()
+      const shouldBeUndefined = await requestHandler.authenticate()
+      await assert.strictEqual(shouldBeUndefined, undefined)
+      scope.done()
+    })
+    it('throws an error after failed authentication', async () => {
+      const scope = BuildServerResponses()
+        .withUnsuccessfulAuthentication()
+        .build()
+      await throwsUnauthorizedException(requestHandler.authenticate())
+      scope.done()
+    })
+  })
+  describe('#get', () => {
+    it('throws an error in case of an error response different from 401', async () => {
+      const errorMessage = 'Failed as intended'
+      const scope = BuildServerResponses()
+        .withUnsuccessfulGetRequest(uri, errorMessage)
+        .build()
+      await throwsException(requestHandler.get(uri), wrapInErrorMessage(errorMessage))
+      scope.done()
+    })
+    it('throws an error if request was answered with status code 401 two times', async () => {
```

Comment:
I don't know if I did something wrong but the last 3 tests from get, delete, update, create and the last 2 from reauthenticate fail for me.